tools/perf/scripts/python/power-statistics.py

```diff
@@ -1319,25 +1343,65 @@ def epilogue(self, fd):
         fd.close()
 
 
-    def parse_and_print_sys_fs_for_residency(self):
-        fd = self.prologue("C-State Idle Residency", file_ext=".json")
-        parsed_proc_data =  {'cpu0': {"residency": 1000}, 'cpu1': {"residency": 10000}}
-        print(json.dumps(parsed_proc_data, indent=4), file=fd)
-        self.epilogue(fd)
-
-
-    def do_whatver_you_want(self):
-        fd = self.prologue("Idle Governor Events")
+    def get_c_state_db(self):
+        c_state_db = collections.defaultdict(dict)
+        sys_path = "/sys/devices/system/cpu/"
+        cpu_dirs = [d for d in os.listdir(sys_path) if re.match(r"^cpu\d+$", d)]
+
+        for cpu_dir in sorted(cpu_dirs, key=lambda x: int(x[3:])):
+            for state in sorted(os.listdir(sys_path + cpu_dir + "/cpuidle/")):
+                c_state_db[cpu_dir.upper()][state] = {}
+                c_state_path = sys_path + cpu_dir + "/cpuidle/" + state + "/"
+                with open(c_state_path + "name", "r") as name_file, open(c_state_path + "residency", "r") as residency_file:
+                    c_state_db[cpu_dir.upper()][state]["name"] = name_file.read().strip()
+                    c_state_db[cpu_dir.upper()][state]["residency"] = residency_file.read().strip()
+        c_state_db = dict(c_state_db)
+        """
+        # implementation for only used states
+        c_state_db = collections.defaultdict(dict)
+        for cpu, c_state in cpu_c_state_pairs:
+            c_state_db[f"CPU{cpu}"][f"state{c_state}"] = {}
+            c_state_path = f"/sys/devices/system/cpu/cpu{cpu}/cpuidle/state{c_state}/"
+            with open(c_state_path + "name", "r") as name_file, open(c_state_path + "residency", "r") as residency_file:
+                c_state_db[f"CPU{cpu}"][f"state{c_state}"]["name"] = name_file.read().strip()
+                c_state_db[f"CPU{cpu}"][f"state{c_state}"]["residency"] = residency_file.read().strip()
+        c_state_db = dict(c_state_db)
+        """
+        return c_state_db
+
+
+    def print_idle_states(self):
+        if args.csv:
+            fmt = csv_sep('{},{},{},{},{},{},{}')
+        else:
+            fmt = csv_sep('{:>20}S{:>5}S{:>10}S{:>20}S{:>20}S{:>4}S{:>4}')
```

Comment:
i want to avoid csv specific branches in the code. Can we not simple use the later for both?